Source/ProgressionSystemRuntime/Private/Data/PSWorldSubsystem.cpp

```diff
@@ -211,66 +210,24 @@ void UPSWorldSubsystem::OnGameStateChanged_Implementation(ECurrentGameState Curr
 	}
 }
 
-// Load game from save file or create a new one (does initial load from data table)
-void UPSWorldSubsystem::LoadGameFromSave()
-{
-	// load from data table
-	const UDataTable* ProgressionDataTable = UPSDataAsset::Get().GetProgressionDataTable();
-	if (!ensureMsgf(ProgressionDataTable, TEXT("ASSERT: [%i] %s:\n'ProgressionDataTable' is not valid!"), __LINE__, *FString(__FUNCTION__)))
-	{
-		return;
-	}
-	UMyDataTable::GetRows(*ProgressionDataTable, ProgressionSettingsDataInternal);
-
-	// Check if the save game file exists
-	if (UGameplayStatics::DoesSaveGameExist(UPSSaveGameData::GetSaveSlotName(), UPSSaveGameData::GetSaveSlotIndex()))
-	{
-		SaveGameDataInternal = Cast<UPSSaveGameData>(UGameplayStatics::LoadGameFromSlot(UPSSaveGameData::GetSaveSlotName(), UPSSaveGameData::GetSaveSlotIndex()));
-	}
-	else
-	{
-		// Save file does not exist
-		// do initial load from data table
-
-		SaveGameDataInternal = Cast<UPSSaveGameData>(UGameplayStatics::CreateSaveGameObject(UPSSaveGameData::StaticClass()));
-
-		if (SaveGameDataInternal)
-		{
-			for (const TTuple<FName, FPSRowData>& Row : ProgressionSettingsDataInternal)
-			{
-				SaveGameDataInternal->SetProgressionMap(Row.Key, FPSSaveToDiskData::EmptyData);
-			}
-		}
-	}
-
-	if (!ensureMsgf(SaveGameDataInternal, TEXT("ASSERT: [%i] %hs:\n'SaveGameDataInternal' failed to create!"), __LINE__, __FUNCTION__))
-	{
-		return;
-	}
-
-	SetFirstElementAsCurrent();
-}
-
 // Always set first levels as unlocked on begin play
 void UPSWorldSubsystem::SetFirstElementAsCurrent()
 {
 	FName FirstSaveToDiskRow = GetFirstSaveToDiskRowName();
-	if (!FirstSaveToDiskRow.IsNone())
+
+	// early return if first element is not valid
+	if (FirstSaveToDiskRow.IsNone())
 	{
-		CurrentRowNameInternal = FirstSaveToDiskRow;
-		SaveGameDataInternal->UnlockLevelByName(CurrentRowNameInternal);
+		return;
 	}
-	SaveDataAsync();
-}
 
-// Updates the stars actors for a spot
-void UPSWorldSubsystem::UpdateProgressionActorsForSpot()
-{
-	AddProgressionStarActors();
+	CurrentRowNameInternal = FirstSaveToDiskRow;
+	SaveGameDataInternal->UnlockLevelByName(CurrentRowNameInternal);
```

Comment:
SaveGameDataInternal might be null

Reply:
done https://github.com/h4rdmol/ProgressionSystem/pull/40/commits/98ed8290b74abc23e558081a8a1b02a40bc4390b